models/MARBL_joint_estimation/model_mod.f90

```diff
@@ -0,0 +1,654 @@
+character(len=vtablenamelength) &
+   :: model_state_variables(modelvar_table_height * modelvar_table_width)
+character(len=vtablenamelength) &
+   :: model_parameters(modelparams_table_height * modelparams_table_width)
+
+namelist /model_nml/ template_file, &
+                     ocean_geometry, &
+                     time_step_days, &
+                     time_step_seconds, &
+                     model_state_variables, &
+                     estimate_params, &
+                     model_parameters
+contains
+
+!------------------------------------------------------------------
+!
+! Called to do one time initialization of the model. As examples,
+! might define information about the model size or model timestep.
+! In models that require pre-computed static data, for instance
+! spherical harmonic weights, these would also be computed here.
+
+subroutine static_init_model()
+
+integer :: iunit, io
+character(len=vtablenamelength) :: variable_table(modelvar_table_height, modelvar_table_width), &
+                                   param_table(modelparams_table_height, modelparams_table_width)
+integer :: state_qty_list(modelvar_table_height), &
+           param_qty_list(modelparams_table_height)
+real(r8):: state_clamp_vals(modelvar_table_height, 2), &
+           param_clamp_vals(modelparams_table_height, 2)
+logical :: update_var_list(modelvar_table_height), &
+           update_param_list(modelparams_table_height)
+
+module_initialized = .true.
+
+! Print module information to log file and stdout.
+call register_module(source)
+
+! Read values from the namelist
+
+call find_namelist_in_file("input.nml", "model_nml", iunit)
+read(iunit, nml = model_nml, iostat = io)
+
+call check_namelist_read(iunit, io, "model_nml")
+
+! Record the namelist values used for the run 
+if (do_nml_file()) write(nmlfileunit, nml=model_nml)
+if (do_nml_term()) write(     *     , nml=model_nml)
+
+! This time is both the minimum time you can ask the model to advance
+! (for models that can be advanced by filter) and it sets the assimilation
+! window.  All observations within +/- 1/2 this interval from the current
+! model time will be assimilated. If this is not settable at runtime 
+! feel free to hardcode it and remove from the namelist.
+assimilation_time_step = set_time(time_step_seconds, &
+                                  time_step_days)
+
+! setting up the DART state vector
+call verify_state_variables(model_state_variables, nfields, variable_table, &
+                            state_qty_list, state_clamp_vals, update_var_list)
+
+    state_dom_id = add_domain(template_file(1), nfields, &
+                              var_names = variable_table(1:nfields, 1), &
+                              kind_list = state_qty_list(1:nfields), &
+                              clamp_vals = state_clamp_vals, &
+                              update_list = update_var_list(1:nfields))
+
+! setting up the DART parameter vector
+if(estimate_params) then
```

Comment:
It's been a while since I was developing the parameter-only version, but as I remember, that setup still needs a separate domain for state variables. The reasoning here was that the forward operator in the parameter-only case involves running MARBL to produce climatological averages, and this operation cannot be coded into DART. Maintaining a separate domain for state variables allows us to compute those averages externally, so that DART can read them as identity observations from a "climatology" component of the state vector. That said, it might be better to rename that component within the model-mod, so that instead of `state_dom_id` we have something like `climatology_dom_id`.